swayfmt/src/items/item_fn/mod.rs

```diff
@@ -35,10 +36,18 @@ impl Format for ItemFn {
                 let body = self.body.get();
                 if !body.statements.is_empty() || body.final_expr_opt.is_some() {
                     Self::open_curly_brace(formatted_code, formatter)?;
+                    formatter.shape.block_indent(&formatter.config);
                     body.format(formatted_code, formatter)?;
                     Self::close_curly_brace(formatted_code, formatter)?;
                 } else {
-                    write!(formatted_code, " {{}}")?;
+                    Self::open_curly_brace(formatted_code, formatter)?;
+                    let range: Range<usize> = self.span().into();
+                    let comments = formatter.comment_map.comments_between(&range);
+                    if has_comments(comments) {
+                        formatter.shape.block_indent(&formatter.config);
+                        maybe_write_comments_from_map(formatted_code, range, formatter)?;
+                    }
```

Comment:
Honestly, the logic to check if a range has comments, and then write the comments + some other logic, could probably be better written. The use case of having to check if comments exist before writing them has come up twice now, because there are times when we might need to write a newline or do an indent before writing the comments (like here).